src/test/lib/utils/meta_tables/meta_temporary_memory_usage_table_test.cpp

```diff
@@ -0,0 +1,110 @@
+ protected:
+  std::shared_ptr<AbstractMetaTable> _meta_temporary_memory_usage_table;
+
+  void SetUp() override {
+    _meta_temporary_memory_usage_table = std::make_shared<MetaTemporaryMemoryUsageTable>();
+  }
+
+  void TearDown() override {
+    Hyrise::reset();
+  }
+
+  const std::shared_ptr<Table> generate_meta_table(const std::shared_ptr<AbstractMetaTable>& table) const {
+    return table->_generate();
+  }
+};
+
+TEST_F(MetaTemporaryMemoryUsageTest, IsImmutable) {
+  EXPECT_FALSE(_meta_temporary_memory_usage_table->can_insert());
+  EXPECT_FALSE(_meta_temporary_memory_usage_table->can_update());
+  EXPECT_FALSE(_meta_temporary_memory_usage_table->can_delete());
+}
+
+TEST_F(MetaTemporaryMemoryUsageTest, TableName) {
+  EXPECT_EQ(_meta_temporary_memory_usage_table->name(), "temporary_memory_usage");
+}
+
+TEST_F(MetaTemporaryMemoryUsageTest, EmptyTableGeneration) {
+  const auto expected_table =
+      Table::create_dummy_table(TableColumnDefinitions{{"operator_type", DataType::String, false},
+                                                       {"operator_data_structure", DataType::String, false},
+                                                       {"timestamp", DataType::Long, false},
+                                                       {"memory_consumption_in_bytes", DataType::Long, false}});
+  EXPECT_TABLE_EQ_UNORDERED(generate_meta_table(_meta_temporary_memory_usage_table), expected_table);
+}
+
+TEST_F(MetaTemporaryMemoryUsageTest, TableGeneration) {
+  // Record some allocations and deallocations.
+  auto& memory_resource_manager = Hyrise::get().memory_resource_manager;
+  memory_resource_manager.enable_temporary_memory_tracking();
+  auto memory_resource_1 = memory_resource_manager.get_memory_resource(OperatorType::Mock, "my_data_structure_1");
+  auto memory_resource_2 = memory_resource_manager.get_memory_resource(OperatorType::Mock, "my_data_structure_2");
+  auto allocated_memory_pointer_1 = memory_resource_1->allocate(10);
+  auto allocated_memory_pointer_2 = memory_resource_1->allocate(20);
+  auto allocated_memory_pointer_3 = memory_resource_2->allocate(5);
+  memory_resource_1->deallocate(allocated_memory_pointer_1, 10);
+  memory_resource_1->deallocate(allocated_memory_pointer_2, 20);
+  memory_resource_2->deallocate(allocated_memory_pointer_3, 5);
+
+  const auto meta_table = generate_meta_table(_meta_temporary_memory_usage_table);
+
+  ASSERT_EQ(meta_table->row_count(), 6);
+
+  const auto timestamp_ns = std::chrono::nanoseconds{std::chrono::steady_clock::now().time_since_epoch()}.count();
+
+  // Check individual rows (we can't simply compare to a pre-built table because the timestamp values will vary between
+  // test runs).
+  const auto row_0 = meta_table->get_row(0);
+  EXPECT_EQ(row_0[0], AllTypeVariant{pmr_string{"Mock"}});
+  EXPECT_EQ(row_0[1], AllTypeVariant{pmr_string{"my_data_structure_1"}});
+  // Recorded timestamp should be within the last second, but before the table was generated.
```

Comment:
That's 60s, right? So the comment needs to be updated. Although I think 1s should be sufficient.